### PR TITLE
Various fixes to restore FreeBSD support

### PR DESCRIPTION
**Pull request by quantum5**

### CODE REVIEW

File: testsuite/rust_generator/tests/AC/test.yml [hunk 1]
```diff
@@ -1,4 +1,5 @@
 language: PY3
+executors: [RUST]
```
Comment (kiritofeng) [inline testsuite/rust_generator/tests/AC/test.yml:2]: Can we rename this to `required_auxiliary_executors` so it's clear what this is for?

File: dmoj/utils/cpp_demangle.py [hunk 2]
```diff
@@ -1,13 +1,23 @@
...
+
+libstdcxx: Any = None if libstdcxx_path is None else CDLL(libstdcxx_path)
+libcxx: Any = None if libcxx_path is None else CDLL(libcxx_path)
 
 # Signature: char *__cxa_demangle(const char *mangled_name, char *output_buffer, size_t *length, int *status)
-__cxa_demangle = libstdcxx.__cxa_demangle
+try:
+    __cxa_demangle = libstdcxx.__cxa_demangle
+except AttributeError:
+    __cxa_demangle = libcxx.__cxa_demangle
```
Comment (kiritofeng) [inline dmoj/utils/cpp_demangle.py:19]: Do we need to do anything here if this also fails? (e.g. `libcxx` is `None`?)

**Review by kiritofeng — COMMENTED** [review]
Otherwise LGTM

File: dmoj/utils/cpp_demangle.py (hunk 2 above)
Comment (quantum5) [reply dmoj/utils/cpp_demangle.py:19]: You are screwed then, it'll crash.

File: testsuite/rust_generator/tests/AC/test.yml (hunk 1 above)
Comment (quantum5) [reply testsuite/rust_generator/tests/AC/test.yml]: Renamed.

**Review by kiritofeng — APPROVED** [review]
LGTM

**Pull request merged**